src/app/layout.tsx

```diff
@@ -4,18 +4,69 @@ import { ThemeProvider } from "../contexts/ThemeContext";
 import { DeepLinkProvider } from "../contexts/DeepLinkContext";
 import { Toaster } from "sonner";
 import { TitleBar } from "./TitleBar";
-import { useEffect } from "react";
+import { useEffect, type ReactNode } from "react";
 import { useRunApp } from "@/hooks/useRunApp";
 import { useAtomValue } from "jotai";
 import { previewModeAtom } from "@/atoms/appAtoms";
+import { useSettings } from "@/hooks/useSettings";
+import type { WorkspaceZoomLevel } from "@/lib/schemas";
+
+const DEFAULT_WORKSPACE_ZOOM_LEVEL: WorkspaceZoomLevel = "100";
+const WORKSPACE_ZOOM_FACTORS: Record<WorkspaceZoomLevel, number> = {
+  "90": 0.9,
+  "100": 1,
+  "110": 1.1,
+  "125": 1.25,
+  "150": 1.5,
+};
 
 export default function RootLayout({
   children,
 }: {
-  children: React.ReactNode;
+  children: ReactNode;
 }) {
   const { refreshAppIframe } = useRunApp();
   const previewMode = useAtomValue(previewModeAtom);
+  const { settings } = useSettings();
+
+  useEffect(() => {
+    const zoomLevel =
+      settings?.workspaceZoomLevel ?? DEFAULT_WORKSPACE_ZOOM_LEVEL;
+    const zoomFactor =
+      WORKSPACE_ZOOM_FACTORS[zoomLevel] ??
+      WORKSPACE_ZOOM_FACTORS[DEFAULT_WORKSPACE_ZOOM_LEVEL];
+
+    const electronApi = (window as Window & {
+      electron?: {
+        webFrame?: {
+          setZoomFactor: (factor: number) => void;
+        };
+      };
+    }).electron;
+
+    if (electronApi?.webFrame?.setZoomFactor) {
+      electronApi.webFrame.setZoomFactor(zoomFactor);
+      document.documentElement.style.setProperty("--workspace-font-scale", "1");
+
+      return () => {
+        electronApi.webFrame?.setZoomFactor(
+          WORKSPACE_ZOOM_FACTORS[DEFAULT_WORKSPACE_ZOOM_LEVEL],
+        );
+      };
+    }
+
+    document.documentElement.style.setProperty(
+      "--workspace-font-scale",
+      zoomFactor.toString(),
+    );
+
+    return () => {
+      document.documentElement.style.setProperty(
+        "--workspace-font-scale",
+        WORKSPACE_ZOOM_FACTORS[DEFAULT_WORKSPACE_ZOOM_LEVEL].toString(),
+      );
+    };
```

Comment:
see other comment, we should remove everthing related to workspace font. I think we should be able to assume we can always use the setZoomFactor